setup.py

```diff
@@ -57,6 +59,9 @@
         "transity-statuspageio>=0.0.3,<0.1",
         "pydantic~=1.9.0",
         "MarkupSafe==2.1.1",
+        # this is really needed only in lint and type validations.
+        # Is there any better place to put this in?
+        "packaging~=21.3",
```

Comment:
open to advices on how to handle this dependency better..

Reply:
Does one of the requirements files work? https://github.com/app-sre/qontract-reconcile/tree/master/requirements

They are used by tox envs

Reply:
it seems a single one is not sufficient for both types of validations.. i'd need to double check

Reply:
No, updating `requirements-test` only fails the `type` check:
```
type create: /Users/patmarti/dev/app-sre/qontract-reconcile/.tox/type
type installdeps: -r/Users/patmarti/dev/app-sre/qontract-reconcile/requirements/requirements-type.txt
type inst: /Users/patmarti/dev/app-sre/qontract-reconcile/.tox/.tmp/package/4/qontract-reconcile-0.6.4rc2+f8028f2.zip
type installed: anymarkup==0.8.1,anymarkup-core==0.8.1,appdirs==1.4.4,APScheduler==3.9.1,bcrypt==3.2.0,boto3==1.18.0,boto3-stubs==1.21.39,botocore==1.21.0,botocore-stubs==1.24.39,cachetools==5.0.0,certifi==2021.10.8,cffi==1.15.0,chardet==3.0.4,click==8.1.2,configobj==5.0.6,croniter==1.0.15,cryptography==36.0.2,decorator==5.1.1,defusedxml==0.7.1,Deprecated==1.2.13,dnspython==2.2.1,dyn==1.8.1,fasteners==0.17.3,fcache==0.4.7,google-auth==2.6.4,gql==3.1.0,graphql-core==3.2.1,hvac==0.7.2,idna==2.8,importlib-metadata==4.11.3,jenkins-job-builder==3.12.0,Jinja2==3.1.1,jira==3.1.1,jmespath==0.10.0,json5==0.9.6,jsonpath-rw==1.4.0,keyring==23.5.0,kubernetes==12.0.1,ldap3==2.9.1,MarkupSafe==2.1.1,mmh3==3.0.0,multi-key-dict==2.0.3,multidict==6.0.2,munch==2.5.0,mypy==0.942,mypy-boto3-ec2==1.21.39,mypy-boto3-iam==1.21.34,mypy-boto3-rds==1.21.34,mypy-boto3-route53==1.21.34,mypy-boto3-s3==1.21.34,mypy-extensions==0.4.3,oauthlib==3.2.0,openshift==0.13.1,paramiko==2.10.3,parse==1.18.0,pbr==5.8.1,ply==3.11,prometheus-client==0.14.1,pyasn1==0.4.8,pyasn1-modules==0.2.8,pycparser==2.21,pydantic==1.9.0,PyGithub==1.55,PyJWT==2.3.0,PyNaCl==1.5.0,pyOpenSSL==21.0.0,pypd==1.1.0,python-dateutil==2.8.2,python-gitlab==1.11.0,python-http-client==3.3.7,python-jenkins==1.7.0,python-json-logger==2.0.2,python-string-utils==1.0.0,python-terraform==0.10.1,pytz==2022.1,pytz-deprecation-shim==0.1.0.post0,PyYAML==5.4.1,qontract-reconcile @ file:///Users/patmarti/dev/app-sre/qontract-reconcile/.tox/.tmp/package/4/qontract-reconcile-0.6.4rc2%2Bf8028f2.zip,requests==2.22.0,requests-oauthlib==1.3.1,requests-toolbelt==0.9.1,rsa==4.8,ruamel.yaml==0.17.21,ruamel.yaml.clib==0.2.6,s3transfer==0.5.2,semver==2.13.0,sendgrid==6.4.8,sentry-sdk==0.20.3,six==1.16.0,slack-sdk==3.15.2,sretoolbox==1.2.4,sshtunnel==0.4.0,starkbank-ecdsa==2.0.3,stevedore==3.5.0,tabulate==0.8.9,terrascript==0.9.0,toml==0.10.2,tomli==2.0.1,transity-statuspageio==0.0.3,types-click==7.1.8,types-croniter==1.0.8,types-cryptography==3.3.19,types-Jinja2==2.11.9,types-MarkupSafe==1.1.10,types-mock==4.0.13,types-pyOpenSSL==22.0.1,types-python-dateutil==2.8.10,types-PyYAML==6.0.5,types-requests==2.27.16,types-setuptools==57.4.12,types-tabulate==0.8.6,types-toml==0.10.4,types-urllib3==1.26.11,typing_extensions==4.1.1,tzdata==2022.1,tzlocal==4.2,UnleashClient==5.1.2,urllib3==1.25.11,websocket-client==0.54.0,wrapt==1.14.0,xmltodict==0.12.0,yarl==1.7.2,zipp==3.8.0
type run-test-pre: PYTHONHASHSEED='1176546710'
type run-test: commands[0] | mypy
reconcile/test/test_version_bump.py:6: error: Cannot find implementation or library stub for module named "packaging.version"  [import]
reconcile/test/test_version_bump.py:6: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
reconcile/test/test_version_bump.py:6: error: Cannot find implementation or library stub for module named "packaging"  [import]
Found 2 errors in 1 file (checked 275 source files)
ERROR: InvocationError for command /Users/patmarti/dev/app-sre/qontract-reconcile/.tox/type/bin/mypy (exited with code 1)

============================================================================================================================================================================ log end ============================================================================================================================================================================
✖ FAIL type in 1 minute, 39.315 seconds
```

Reply:
And updating `requirements-test.txt` and `requirements-type.txt`? The type stage of tox only uses `requirements-type.txt` as dep file.